### PR TITLE
feat(village): implement get village detail by id

### PR DESCRIPTION
**Pull request by AldiNFitrah**
### Response Example
image

cc @jabardigitalservice/jds-backend 

### Evidence
- title: Implement get village detail by id
- project: Desa Digital
- participants: @ayocodingit @firmanJS @tukangremot

### CODE REVIEW

File: src/modules/villages/village_handler.ts [hunk 1]
```diff
@@ -8,7 +9,30 @@ const router = express.Router()
...
   })
 
+router.get(
+  '/v1/villages/:id',
+  async (req: Request<never, never, never, Entity.RequestQuery>, res: Response, next: NextFunction) => {
+    try {
+      res
+        .status(httpStatus.OK)
+        .json({
+          data: await Service.findById(req.params),
```
Comment (ayocodingit) [inline src/modules/villages/village_handler.ts:23]: mapping for response beeter in service, then add response interface

File: src/modules/villages/village_handler.ts [hunk 2]
```diff
@@ -8,7 +9,30 @@ const router = express.Router()
 router.get(
   '/v1/villages/list-with-location',
   async (req: Request<never, never, never, Entity.RequestQuery>, res: Response, next: NextFunction) => {
+    console.log('masuk')
```
Comment (ayocodingit) [inline src/modules/villages/village_handler.ts:12]: please remove log :smile:

File: src/modules/villages/village_handler.ts [hunk 3]
```diff
@@ -8,7 +9,30 @@ const router = express.Router()
...
+  async (req: Request<never, never, never, Entity.RequestQuery>, res: Response, next: NextFunction) => {
+    try {
+      res
+        .status(httpStatus.OK)
+        .json({
+          data: await Service.findById(req.params),
+          meta: {},
+        })
+    } catch (err) {
+      if (err instanceof HttpError) {
```
Comment (ayocodingit) [inline src/modules/villages/village_handler.ts:27]: use next if use catch

example:
`next(err)`

File: src/modules/villages/village_service.ts [hunk 4]
```diff
@@ -50,4 +52,28 @@ export namespace Village {
 
     return data
   }
+
+  export const findById = async ({ id }: { id: string }): Promise<Entity.FindById> => {
+    const item = await Repository.findById(id)
+
+    if (item === undefined) {
```
Comment (ayocodingit) [inline src/modules/villages/village_service.ts:59]: shortcut use example `if (!item) throw new HttpError(httpStatus.NOT_FOUND, 'village with id ${id} does not exits')`

File: src/modules/villages/village_handler.ts (hunk 2 above)
Comment (AldiNFitrah) [reply src/modules/villages/village_handler.ts:12]: ah forget to remove it wkwkwk

File: src/modules/villages/village_handler.ts [hunk 5]
```diff
@@ -11,4 +12,17 @@ router.get(
     res.status(httpStatus.OK).json(await Service.findAllWithLocation(req.query))
   })
 
+router.get(
+  '/v1/villages/:id',
+  async (req: Request<never, never, never, Entity.RequestQuery>, res: Response, next: NextFunction) => {
```
Comment (ayocodingit) [inline src/modules/villages/village_handler.ts:17]:
```suggestion
  async (req: Request, res: Response, next: NextFunction) => {
```

please remove if not use

File: src/modules/villages/village_service.ts [hunk 6]
```diff
@@ -50,4 +52,32 @@ export namespace Village {
...
   }
+
+  export const findById = async ({ id }: { id: string }): Promise<Entity.ResponseFindById> => {
+    const item = await Repository.findById(id)
+
+    if (!item) {
+      throw new HttpError(httpStatus.NOT_FOUND, `village with id ${id} does not exits`)
+    }
+
+    const data = {
```
Comment (ayocodingit) [inline src/modules/villages/village_service.ts:63]: please add interface ResponseFindById 
`
{
data: {},
meta: {}
}
`

then make variable result implement interface ResponseFindById

File: src/modules/villages/village_handler.ts (hunk 5 above)
Comment (AldiNFitrah) [reply src/modules/villages/village_handler.ts:17]: turns out I need the params, I will change it

File: src/modules/villages/village_service.ts (hunk 6 above)
Comment (AldiNFitrah) [reply src/modules/villages/village_service.ts:63]: https://github.com/jabardigitalservice/dedi-core-service/blob/4cc9b6270711c3aee52cc8e1147e07a972e4964f/src/modules/villages/village_entity.ts#L60-L63

this one?

File: src/modules/villages/village_handler.ts [hunk 7]
```diff
@@ -11,4 +12,17 @@ router.get(
     res.status(httpStatus.OK).json(await Service.findAllWithLocation(req.query))
```
Comment (firmanJS) [inline src/modules/villages/village_handler.ts:11]: this is bad code, it should be defined first, into const

example ` 
const result: object | any = await Service.findAllWithLocation(req.query)`

or you can't set in interface

if you have already followed the existing base code, it should be replaced first before it becomes a lot of files

cc @ayocodingit

File: src/modules/villages/village_service.ts (hunk 6 above)
Comment (ayocodingit) [reply src/modules/villages/village_service.ts:63]: yups :+1:

Comment (AldiNFitrah) [reply src/modules/villages/village_service.ts:63]: it's already implemented then

File: src/modules/villages/village_handler.ts (hunk 7 above)
Comment (firmanJS) [reply src/modules/villages/village_handler.ts:11]: jadi di `.json()` lebih baik di define terlebih dahulu

contoh const result = bla.bla nanti di `.json(result)`

pros nya apa : 
lebih enak di baca dan sesuai standar juga
cons : 
mungkin agak ribet harus di define terlebih dahulu, akan tetapi tidak sesuai standar TS

Comment (ayocodingit) [reply src/modules/villages/village_handler.ts:11]: iyups setuju setuju, thank mang @firmanJS , masih mencari mana yang bagus nya

Comment (firmanJS) [reply src/modules/villages/village_handler.ts:11]: gas gas next pr naikin dulu fitur ini :runner:

**Review by tukangremot — APPROVED** [review]

**Review by ayocodingit — APPROVED** [review]
good job :tada: :tada:

**Pull request merged**